src/pages-conditional/dapps.tsx

```diff
@@ -1242,7 +1242,7 @@ const DappsPage = ({
           ))}
         </StyledCardGrid>
       </Content>
-      <FullWidthContainer ref={explore}>
```

Comment:
Doesn't seems like `FullWidthContainer` has used this prop at all as well as the parent components of it, and the `explore` here was only used to grab the DOM. LMK if this is not expected!

Reply:
Got it. So there's an issue explained in 8ac930142fd1d137657bd697d54df42f3447657c but the gist is that when I type it with:
```ts
const FullWidthContainer = styled(Page)<{
  ref: React.RefObject<HTMLElement>
}>`
```
It will complain that it wont be able to match the type of `RefObject`, so I did some digging and found out that this particular property of this type is ignored [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e358cde5edeaa8156bc5b9086dd9f5443c2cadc7/types/react/index.d.ts#L818)

I am not too sure what the resolution to this could be. For now, I changed the name from `ref` to `scrollRef`, and the typing will work

Reply:
There is no need to define the `ref` prop here. That prop is already defined by React. It is not a custom prop.

So,
1. We should keep `const FullWidthContainer = styled(Page)` as before
2. Keep using `ref` in `<FullWidthContainer ref={explore}>`
3. One thing I noticed is that we will need to change the ref type, from `HTMLElement` to `HTMLDivElement`, in `useRef`

Reply:
Ohhh I think that's why I was unable to type it since `ref` got a type mismatch from `HTMLElement` instead of the correct type `HTMLDivElement`, got a bit confused. Thanks for clearing it up! Changes updated